databroker/broker.py

```diff
@@ -233,6 +234,25 @@ def __getitem__(self, key):
         """Do-What-I-Mean slicing"""
         return search(key, self.mds)
 
+    def __getattr__(self, key):
+        try:
+            query = self.aliases[key]
+        except KeyError:
+            raise AttributeError(key)
+        if callable(query):
+            query = query()
+        return self(**query)
+
+    def alias(self, key, **query):
```

Comment:
Oops I'll add a docstring.

Get Outlook for iOS<https://aka.ms/o0ukef>




On Thu, Sep 1, 2016 at 7:27 PM -0400, "K Lauer" <notifications@github.com<mailto:notifications@github.com>> wrote:


In databroker/broker.py<https://github.com/NSLS-II/databroker/pull/77#discussion_r77271961>:

> @@ -233,6 +234,25 @@ def __getitem__(self, key):
>          """Do-What-I-Mean slicing"""
>          return search(key, self.mds)
>
> +    def __getattr__(self, key):
> +        try:
> +            query = self.aliases[key]
> +        except KeyError:
> +            raise AttributeError(key)
> +        if callable(query):
> +            query = query()
> +        return self(**query)
> +
> +    def alias(self, key, **query):


Too late to bikeshed, darn. alias being ambiguously as a verb/noun and there being no docstring, I was confused to where this was going...

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/NSLS-II/databroker/pull/77/files/77eaf21cde384499fb2ca510be503f92dea7fe6f#r77271961>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACLIrmnVxfs0NYriEmDn2sCcNAy5D6Fjks5ql19QgaJpZM4Jy89b>.